src/transactions/feed/JumpstartTransactionDetailsScreen.tsx

```diff
@@ -104,47 +137,88 @@ function JumpstartTransactionDetailsScreen({ route }: Props) {
       if (!isDeposit) {
         return
       }
-      return await getClaimDataAndStatus(
+
+      const { beneficiary, index, claimed } = await getClaimDataAndStatus(
         jumpstartContractAddress as Address,
-        transaction.networkId,
+        networkId,
         transaction.transactionHash as Hash
       )
+
+      if (claimed) {
+        return { claimed: true }
+      }
+
+      const reclaimTx: TransactionRequest = {
+        from: walletAddress as Address,
+        to: jumpstartContractAddress as Address,
+        value: BigInt(0),
+        data: encodeFunctionData({
+          abi: walletJumpstart.abi,
+          functionName: 'reclaimERC20',
+          args: [beneficiary, BigInt(index)],
+        }),
+      }
+
+      const preparedTransactions = await prepareTransactions({
+        feeCurrencies,
+        baseTransactions: [reclaimTx],
+      })
+
+      const resultType = preparedTransactions.type
+      switch (resultType) {
+        case 'need-decrease-spend-amount-for-gas': // fallthrough on purpose
+        case 'not-enough-balance-for-gas':
+          return {
+            claimed: false,
+            error: true,
+          }
+        case 'possible':
+          return {
+            preparedTransaction: preparedTransactions.transactions[0],
+            claimed: false,
+          }
+      }
     },
     [],
     {
+      onSuccess(result) {
+        if (result) {
+          if (result.preparedTransaction) {
+            setReclaimTx(getSerializablePreparedTransaction(result.preparedTransaction))
+          }
+          if (result.error) {
+            setError(new Error('Not enough balance for gas'))
+          }
+        }
+      },
       onError: (error) => {
-        // TODO: Show error message in the UI
+        setError(new Error('Failed to fetch escrow data'))
         Logger.error(TAG, 'Failed to fetch escrow data', error)
       },
     }
   )
 
-  const onReclaimPress = () => {
-    if (!fetchClaimData.result) {
-      Logger.error(TAG, 'No escrow data found when trying to reclaim')
-      return
-    }
+  const resetState = () => {
+    setReclaimTx(null)
+    setError(null)
+    void fetchClaimData.execute()
```

Comment:
nit: i don't think we need to refetch here, because the button CTA doesn't indicate that it will try again?

Reply:
Yeah, I know the CTA is only dismiss. But we say in the copy "Try again", but what do we offer if the fetch failed?
The user has to go back to the feed and enter again to try? I think this refetch is useful in that case.
The other option would be to leave the button disabled.